src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -599,27 +597,33 @@ public bool Remove(TKey key)
                 int hashCode = comparer.GetHashCode(key) & 0x7FFFFFFF;
                 int bucket = hashCode % buckets.Length;
                 int last = -1;
-                for (int i = buckets[bucket]; i >= 0; last = i, i = entries[i].next)
+                int i = buckets[bucket];
+                while (i >= 0)
```

Comment:
You mean the change from `for` to `while`? It's because `entries[i]` was used in the iterator expression of the `for` statement and there `entry` is not available.

I could have kept `for` and only move the iterator expression to the end of the `for` body. But it's not a typical `for` anyway so I changed it to `while`.